drivers/SerialBase.cpp

```diff
@@ -73,9 +76,11 @@ void SerialBase::attach(Callback<void()> func, IrqType type) {
     // Disable interrupts when attaching interrupt handler
     core_util_critical_section_enter();
     if (func) {
+        sleep_manager_lock_deep_sleep();
```

Comment:
Attach could also be called to replace a callback of the same type. If this is the case then deep sleep shouldn't be locked a second time.